typing/parmatch.ml

```diff
@@ -886,30 +857,38 @@ let rec get_variant_constructors env ty =
     end
   | _ -> fatal_error "Parmatch.get_variant_constructors"
 
-(* Sends back a pattern that complements constructor tags all_tag *)
-let complete_constrs constr all_tags =
+module ConstructorSet = Set.Make(struct
+  type t = constructor_description
+  let compare c1 c2 = String.compare c1.cstr_name c2.cstr_name
+end)
```

Comment:
This is essentially a `constructor_description String.Map.t` or even a `String.Set.t`, isn't it?

Reply:
I think that the current approach is nicer, because `string` is not an informative type. (Originally Nicolas had a `Set.Make(String).t` instead of a `ConstructorSet.t`, and I asked him to change this.) For example in the PR `complete_constrs` has an argument that changes from `constructor_tag list` to `constructor_description list`, you don't want to have a `string list` in there.

Reply:
I agree that it is mostly a question of bikeshedding.